src/Microsoft.AspNet.Server.Kestrel/Http/MessageBody.cs

```diff
@@ -326,6 +349,17 @@ private static bool TakeChunkedLine(SocketInput baton, ref int chunkSizeOut)
                 }
             }
 
+            protected override void Dispose(bool disposing)
+            {
+                if (_inputLength > 0)
```

Comment:
What if we have keep alive connection? Would this run several times for every chunk?